server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java

```diff
@@ -88,12 +86,12 @@ public CompletableFuture<BatchIterator<Row>> getIterator(Role currentUser,
                                                              TransactionContext txnCtx,
                                                              List<Symbol> collect) {
         SessionSettings sessionSettings = txnCtx.sessionSettings();
-        Map<String, Object> userOptions = server.users().get(sessionSettings.userName());
+        Settings userOptions = server.users().get(sessionSettings.userName());
         if (userOptions == null) {
-            userOptions = Map.of();
+            userOptions = Settings.EMPTY;
         }
-        String user = DataTypes.STRING.implicitCast(userOptions.get("user"));
-        String password = DataTypes.STRING.implicitCast(userOptions.get("password"));
+        String user = userOptions.get("user");
+        String password = userOptions.get("password");
```

Comment:
this should go through the setting definition. Same as the remaining options (e.g. urlSetting)

Reply:
Once https://github.com/crate/crate/pull/15606 is merged, the settings will be available to use here.